go/arrow/array/util_test.go

```diff
@@ -430,6 +430,10 @@ func TestArrRecordsJSONRoundTrip(t *testing.T) {
 			// float weirdness due to their size, so smaller tests will work fine.
 			continue
 		}
+		if k == "run_end_encoded" {
+			// unmarshalling json to run_end_encoded not yet implemented
```

Comment:
Do you plan to do it in this PR?

Reply:
I hadn't planned on doing it in this PR yet, there's some discussion that should probably happen on whether or not a JSON representation of run-end-encoded values should be expanded out (like dictionaries) or if there should be an explicit run-end-encoded JSON format (would require some specialization for nested cases since they wouldn't be able to just use `.Len` on the array)

Reply:
after chatting with @felipecrv and coming to a decision on whether or not we wanted to expand the array (like how JSON marshalling of dictionary arrays expands the array), I've implemented JSON unmarshalling and added it to this PR, removing this conditional check.

Reply:
@pitrou this is regular JSONL files, not the JSON integration format.